Shared/src/com/pacoapp/paco/shared/util/QueryPreprocessor.java

```diff
@@ -276,19 +292,25 @@ public void visit(InExpression inExpression) {
     if (le != null) {
       if (le instanceof Column) {
         Column leftColumn = (Column) le;
-        if (modifyDateToUTC && requestedDateColumns.contains(leftColumn.getColumnName())) {
+        String leftColName = leftColumn.getColumnName();
+        if (requestedDateColumns.contains(leftColName)) {
           if (ril instanceof ItemsList) {
             ExpressionList expList = (ExpressionList) ril;
             List<Expression> elList = expList.getExpressions();
             List<Expression> newUtcList = Lists.newArrayList();
-            SimpleDateFormat sdfLocal = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+            String utcFormattedDate = null;
             for (Expression expr : elList) {
               if (expr instanceof StringValue) {
                 try {
-                  Date dt = sdfLocal.parse(((StringValue) expr).getValue());
-                  Date utcDate = TimeUtil.convertToUTC(dt, timeZone);
-                  String utcFormattedDate = sdfLocal.format(utcDate);
-                  newUtcList.add(new StringValue(utcFormattedDate));
+                  if (modifyDateToUTC) {
```

Comment:
It looks like this block and the one at line 437 are the same. If so, please extract a method to use in each place.